tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -70,7 +70,23 @@ enum FilterFailureReason
   OutTheBack,
   /// The frame_id on the message is empty
   EmptyFrameID,
+  /// Max enum value for iteration, keep it at the end of the enum
```

Comment:
Consider adding a comment at the top of the enum asking people *not* to assign number to those, as this would break this mechanism.